ui/dashboards/src/components/GridLayout/GridLayout.tsx

```diff
@@ -88,3 +108,15 @@ export function GridLayout(props: GridLayoutProps) {
     </GridContainer>
   );
 }
+
+/**
+ * Calculates grid item width
+ * @param w number of columns the grid item spans
+ * @param colWidth the width of each column in px
+ * @returns grid item's width in px
+ */
+const calculateGridItemWidth = (w: number, colWidth: number) => {
+  // 0 * Infinity === NaN, which causes problems with resize contraints
+  if (!Number.isFinite(w)) return w;
+  return colWidth * w + Math.max(0, w - 1) * DEFAULT_MARGIN;
```

Comment:
```suggestion
  return Math.round(colWidth * w + Math.max(0, w - 1) * DEFAULT_MARGIN);
```
See comment above. I tested these changes locally and validated that the dom size and the value on the prop match when you swap where the rounding happens.